src/Terminus.php

```diff
@@ -293,8 +294,13 @@ private function configureContainer()
             ->invokeMethod('setSites', ['sites']);
 
         // Tell the command loader to only allow command functions that have a name/alias.
-        $factory = $container->get('commandFactory');
+        $command_cache = new FileStore($this->getConfig()->get('cache_dir') . '/commands');
+        $factory = new CachedAnnotatedCommandFactory();
+        $factory->setDataStore($command_cache);
         $factory->setIncludeAllPublicMethods(false);
+        $factory->setCommandProcessor($container->get('commandProcessor'));
+
+        $container->share('commandFactory', $factory);
```

Comment:
I'm not sure I 100% follow the issue, but your suggestion is certainly cleaner. Changed.